### DISCUSSION TITLE
GPUs for calculation a bad idea?

### DISCUSSION THREAD

**Discussion by cd1188** [Yutaka-Sawada/MultiPar#113] [Q&A] [answered]
Isn't using consumer GPUs for calculations a bad idea?
To avoid calculation errors, professional GPUs always have ECC memory installed.
There was once an article in which scientists complained that the GPUs were calculating incorrectly and then had to admit that they had used consumer gaming cards from Nvidia.

Edit: Sorry, should I have opened a discussion thread instead of a issue?

**Comment by Yutaka-Sawada** [comment, marked as answer]
> Isn't using consumer GPUs for calculations a bad idea?

When it can detect calculation error of GPU, using GPU for calculations is a good idea. My PAR clients include error detection of calculated data. While I implemented it to find memory error on GPU, it will find most calculation error, too. Some users reported memory error at CPU calculation ago. But, I never see such error of CPU or GPU on my PCs yet. It seems that calculation error is rare and depends on hardware environments.

> Sorry, should I have opened a discussion thread instead of a issue?

There is no problem. You might afraid of GPU's calculation (or VRAM) error. I know the risk. When MultiPar detects such error, it will fail by "checksum mismatch". So, users can know what happened on his PC. Because I never see the failure, it doesn't resume process after the error.
